LBHackney-IT/lbh-fss-portal-frontend
# Feature/update Register component

**Description**

Add functionality to Register component:
 - The registration process now consists of a three step form
    - Step 1: name + email + confirm T&Cs
    - Step 2: password + confirm password (at this point a confirmation code is sent)
    - Step 3: email + confirmation code + resend confirmation code
 - Values from step 1 are saved in a cookie, in order to preserve values between sessions 
 - The cookie is read during step 2, if the cookie is unable to be read the cookie is removed and the user is redirected to step 1
  - If a user attempts to access step 2 without a cookie they are redirected to step 1
  - After the cookie has been read during step 2 the values in the cookie are re-checked to ensure they meet the necessary validation criteria (this is incase a user attempts to update the value of the cookie in order to bypass front-end validation rules) 
  - After the user has successfully submitted the form on step 2 they will be sent a confirmation code, this confirmation code is to be input on step 3 of the form 
  - The user has the option to resend the confirmation code on step 3

Note - this pull request does not include tests for this functionality, this will be added in a separate commit

**Test instructions**

- ensure you are able to navigate to unprotected routes (i.e. ensure the mockUser has a role of 'vcso', rather than 'viewer' or 'admin')
- navigate to **/register/step-2**, check you are automatically redirected to **/register/step-1** (this is the expected behaviour and occurs because a cookie has not been set)
- navigate to **/register/step-1** enter the form correctly and click 'Create Account', now open a new session and ensure the values persist in the form, now click 'Create Account' again, enter a valid password and  click 'Create Password', you should receive a message indicating a confirmation code has been sent and redirected to **/register/step-3** with the email field auto-populated, enter in a code and check you are logged in  
- now navigate back to **/register/step-1**, enter the form correctly and click 'Create Account', you will be redirected to **/register/step-2**, now edit the values in the cookie to something which would not pass validation (e.g. email in incorrect format, or name too long), enter in a valid password and click 'Create Password', you should be redirected to the **/register/step-1** and the cookie removed 
 - go through steps 1 and 2, once on step 3 try using the 'Resend confirmation code' code button, also try entering an invalid email address in the email field and again try 'Resend confirmation code' (you should get an error message)

Please review the code in the following files:
  - src/domain/Authentication/Register/RegisterStep1/RegisterStep1.js
  - src/domain/Authentication/Register/RegisterStep1/RegisterStep2.js
  - src/domain/Authentication/Register/RegisterStep1/RegisterStep3.js

**Ticket**

https://app.clickup.com/t/9e5u6q
